zebra-consensus/src/checkpoint/list.rs

```diff
@@ -122,11 +116,9 @@ impl CheckpointList {
 
         // Check that the list starts with the correct genesis block
         match checkpoints.iter().next() {
-            Some((block::Height(0), hash))
-                if (hash == &genesis_hash(Network::Mainnet)
-                    || hash == &genesis_hash(Network::Testnet)) => {}
+            Some((block::Height(0), hash)) if hash == &genesis_hash(network) => {}
             Some((block::Height(0), _)) => {
-                Err("the genesis checkpoint does not match the Mainnet or Testnet genesis hash")?
+                Err("the genesis checkpoint does not match the network genesis hash")?
             }
             Some(_) => Err("checkpoints must start at the genesis block height 0")?,
             None => Err("there must be at least one checkpoint, for the genesis block")?,
```

Comment:
Rather than complicating this code by adding a network parameter, let's move the checks that depend on `network` into `new()`. (And replace the similar check already in `new()`.)